modules/public/views.py

```diff
@@ -1,214 +1,122 @@
-    )
+from urllib.parse import quote_plus
 
 
-def get_pages_details(links):
-    links_result = {}
-
-    if('first' in links):
-        links_result['first'] = convert_limit_offset_to_size_page(
-            links['first']['href']
-        )
-
-    if('last' in links):
-        links_result['last'] = convert_limit_offset_to_size_page(
-            links['last']['href']
-        )
-
-    if('next' in links):
-        links_result['next'] = convert_limit_offset_to_size_page(
-            links['next']['href']
-        )
-
-    if('prev' in links):
-        links_result['prev'] = convert_limit_offset_to_size_page(
-            links['prev']['href']
-        )
-
-    if('self' in links):
-        links_result['self'] = convert_limit_offset_to_size_page(
-            links['self']['href']
+def homepage():
+    featured_snaps = []
+    errors = []
+    try:
+        featured_snaps = logic.normalize_searched_snaps(
+            api.get_featured_snaps()
         )
+    except api.InvalidResponseContent as invalid_response_content:
+        errors = ["Error from server"]
+    except api.ApiErrorResponse as api_error_exception:
+        if api_error_exception.errors:
+            errors = api_error_exception.errors
+        else:
+            errors = ["Unknown error"]
 
-    return links_result
-
-
-def convert_limit_offset_to_size_page(link):
-    url_parsed = urlparse(link)
-    host_url = (
-        "{base_url}"
-        "?q={q}&limit={limit}&offset={offset}"
-    )
-
-    url_queries = parse_qs(url_parsed.query)
-    q = url_queries['q'][0]
-    size = int(url_queries['size'][0])
-    page = int(url_queries['page'][0])
-
-    return host_url.format(
-        base_url=flask.request.base_url,
-        q=q,
-        limit=size,
-        offset=size*(page-1)
-    )
-
-
-def homepage():
     return flask.render_template(
         'index.html',
-        featured_snaps=api.get_featured_snaps()
+        featured_snaps=featured_snaps,
+        errors=errors
     )
 
 
 def store():
+    featured_snaps = []
+    errors = []
+    try:
+        featured_snaps = logic.normalize_searched_snaps(
+            api.get_featured_snaps()
+        )
+    except api.InvalidResponseContent as invalid_response_content:
+        errors = ["Error from server"]
+    except api.ApiErrorResponse as api_error_exception:
+        if api_error_exception.errors:
+            errors = api_error_exception.errors
+        else:
+            errors = ["Unknown error"]
+
     return flask.render_template(
         'store.html',
-        featured_snaps=api.get_featured_snaps(),
-        page_slug='store'
+        featured_snaps=featured_snaps,
+        page_slug='store',
+        errors=errors
     )
 
 
 def snaps():
+    promoted_snaps = []
+    errors = []
+    try:
+        promoted_snaps = logic.normalize_searched_snaps(
+            api.get_promoted_snaps()
+        )
+    except api.InvalidResponseContent as invalid_response_content:
+        errors = ["Error from server"]
```

Comment:
It feels like we're losing information here if all we pass on is "Error from server". We know it's about the content of the response. Couldn't we be slightly more descriptive in the public message?